src/ast/documentation.ts

```diff
@@ -0,0 +1,112 @@
+import { ASTNode, ASTNodeWithChildren } from "./ast_node";
+import { StructuredDocumentation } from "./implementation/meta";
+
+export interface WithPrecedingDocs {
+    documentation?: string | StructuredDocumentation;
+    docString?: string;
```

Comment:
Why do we have 2 fields - `docString` and `documentation`? It feels like having a single `documentation` field should be enough

Reply:
Well, `docString` is defined specifically to store a `string`, when it is set as a value for the property. The `documentation` is the external property that is implemented by **getter**, that works in following order:
- If `docString` **is not** `undefined`, then retuns `docString`.
- If `docString` **is** `undefined`, then check `children` from very beginning and return `StructuredDocumentation` node that is preceeding current one: struct docs source range is out of the parent node source range (in other words they are excluded from parent's source range by compiler).
- If there is none of above applicable - return `undefined`.

Same applies to the `danglingDocumentation` with few exceptions:
- The `children` are checked from very end.
- Then `StructuredDocumentation` node need to be included into parent's source range.

Reply:
Added a bit more documentation to `docString` and `danglingDocString` in https://github.com/ConsenSys/solc-typed-ast/pull/115/commits/872be86fefb3d356d2552929b0a260684ffd6801